Packs/BmcHelixRemedyForce/Integrations/BmcHelixRemedyForce/BMCHelixRemedyforce.py

```diff
@@ -0,0 +1,2883 @@
+                                                    "service_request_definition_params")
+    additional_fields, excluded_fields = get_valid_arguments(args.get("additional_fields"), "additional_fields")
+    if answers and isinstance(safe_load_json(answers), dict):
+        for each_answer in answers:
+            temp: Dict[str, Any] = dict()
+            temp["Values"] = list()
+            temp["QuestionId"] = each_answer
+            temp["Values"].append(answers[each_answer])
+            answers_list.append(temp)
+    elif answers:
+        raise ValueError("{}".format(MESSAGES["INVALID_FORMAT_ERROR"]).format(
+            "service_request_definition_params", "param1=value1; param2=value2"))
+    if additional_fields and isinstance(safe_load_json(additional_fields), dict):
+        invalid_fields = list()
+        for each_field in additional_fields:
+            if each_field in AVAILABLE_FIELD_LIST:
+                invalid_fields.append(each_field)
+        if invalid_fields:
+            raise AttributeError("{}".format(MESSAGES["INVALID_FIELDS_ERROR"]).format(
+                ", ".join(invalid_fields), "additional_fields"))
+    elif additional_fields:
+        raise ValueError("{}".format(MESSAGES["INVALID_FORMAT_ERROR"]).format(
+            "additional_fields", "field_id1=value_1; field_2=value_2"))
+    create_service_request(client, service_request_definition, answers=answers_list,
+                           additional_fields=additional_fields,
+                           excluded_fields=excluded_fields, default_args=args)
+
+
+@logger
+def bmc_remedy_incident_create_command(client, args):
+    """
+    Creates an incident.
+
+    :param client: Client instance
+    :param args: Command arguments
+    :return:
+    """
+    # Request body for create incident api call
+    args = remove_extra_space_from_args(args)
+
+    if "client_id" not in args:
+        raise DemistoException(MESSAGES['EMPTY_REQUIRED_ARGUMENT'].format("client_id"))
+
+    create_request_body = {'Description': args.pop('description', '')}
+
+    # Prepare update request payload and get field names from additional arguments
+    update_payload, fields = get_update_incident_payload(args)
+
+    validate_incident_update_payload(update_payload)
+
+    # Call create incident api
+    api_response = client.http_request('POST', url_suffix=URL_SUFFIX['CREATE_INCIDENT'], json_data=create_request_body)
+
+    create_result = api_response.get('Result')
+
+    if not api_response.get('Success', False) or not create_result or 'Id' not in create_result:
+        raise DemistoException(HR_MESSAGES['CREATE_INCIDENT_FAILURE'].format(
+            api_response.get('ErrorMessage', MESSAGES['UNEXPECTED_ERROR'])))
+
+    try:
+        id_suffix = '/{}'.format(create_result.get('Id', ''))
+        update_api_response = client.http_request('PATCH', URL_SUFFIX['UPDATE_INCIDENT'] + id_suffix,
```

Comment:
why is this being called?

Reply:
BMC service request create API accepts only one argument, that is service request definition.
Now to make it user friendly like BMC UI, we have used Salesforce update API to pass the remaining argument as part of the update request, same applies for incident.
Please let me know if still there are any doubts.